rq/job.py

```diff
@@ -733,4 +732,38 @@ def register_dependency(self, pipeline=None):
             connection.sadd(dependents_key, self.id)
             connection.sadd(self.dependencies_key, dependency_id)
 
+    def dependencies_finished(
+            self,
+            pipeline=None
+    ):
+        """Returns a boolean indicating if all of this jobs dependencies are _FINISHED_
+
+        If a pipeline is passed, all dependencies are WATCHed.
+        """
+
+        pipe = pipeline if pipeline is not None else self.connection
+
+        if pipeline is not None:
+            pipe.watch(*[Job.key_for(as_text(_id))
```

Comment:
I feel like we should have a convenience function to get all dependencies' ID, something like `job.dependencies_ids`?

Reply:
agreed! will do.

Reply:
done in e1674e79d12aef7fb16f91b29b0f9be03c551a7a